browser/src/app/TilesMiddleware.ts

```diff
@@ -259,6 +334,10 @@ class TileManager {
 		if (this.debugDeltas)
 			window.app.console.log('Garbage collect! iter: ' + this.gcCounter);
 
+		// In case garbage collection was forced outside of maybeGarbageCollect, make sure future
+		// garbage collection doesn't happen prematurely.
+		this.gcCounter += (53 - (this.gcCounter % 53)) % 53;
```

Comment:
This change stops garbage collection happening prematurely in the case that it gets forced. It still happens periodically during scrolling of the document, but that's probably ok.